packages/extension-koni-ui/src/hooks/ledger/useLedger.ts

```diff
@@ -221,8 +221,19 @@ export function useLedger (slug?: string, active = true, isSigning = false, forc
       });
   }, [getLedger, ledger, t]);
 
-  const signTransaction = useCallback(async (message: Uint8Array, metadata: Uint8Array, accountOffset?: number, addressOffset?: number, accountOption?: Partial<AccountOptions>): Promise<LedgerSignature> => {
+  const signTransaction = useCallback(async (message: Uint8Array, metadata: Uint8Array, accountOffset?: number, addressOffset?: number, address?: string, accountOption?: Partial<AccountOptions>): Promise<LedgerSignature> => {
     if (ledger) {
+      const addressOnCurrentLedger = await ledger.getAddress(false, accountOffset, addressOffset);
+
+      if (addressOnCurrentLedger.address !== address) {
```

Comment:
Replace with deep compare(`isSameAddress`), `addressOnCurrentLedger` can be in network format